pdr_backend/util/subgraph_predictions.py

```diff
@@ -0,0 +1,211 @@
+    elif filter_mode == FilterMode.PREDICTOOR:
+        where_clause = f"where: {{user_: {{id_in: {json.dumps(filters)}}}, slot_: {{slot_gt: {start_ts}, slot_lt: {end_ts}}}}}"
+
+    while True:
+        query = f"""
+            {{
+                predictPredictions(skip: {offset}, first: {chunk_size}, {where_clause}) {{
+                    id
+                    user {{
+                        id
+                    }}
+                    stake
+                    payout {{
+                        payout
+                        trueValue
+                        predictedValue
+                    }}
+                    slot {{
+                        slot
+                        predictContract {{
+                            id
+                            token {{
+                                id
+                                name
+                                nft{{
+                                    nftData {{
+                                    key
+                                    value
+                                    }}
+                                }}
+                            }}
+                        }}
+                    }}
+                }}
+            }}"""
+
+        # print("Querying subgraph...", query)
+        result = query_subgraph(
+            get_subgraph_url(network),
+            query,
+            timeout=20.0,
+        )
+
+        offset += chunk_size
+
+        if not "data" in result:
+            break
+
+        data = result["data"]["predictPredictions"]
+        if len(data) == 0:
+            break
+
+        for prediction in data:
+            info725 = prediction["slot"]["predictContract"]["token"]["nft"]["nftData"]
+            info = info_from_725(info725)
+            pair_name = info["pair"]
+            timeframe = info["timeframe"]
+            source = info["source"]
+            timestamp = prediction["slot"]["slot"]
+
+            if prediction["payout"] is None:
+                continue
+
+            trueval = prediction["payout"]["trueValue"]
+            payout = float(prediction["payout"]["payout"])
+
+            if trueval is None:
+                continue
+
+            predictedValue = prediction["payout"]["predictedValue"]
+            stake = float(prediction["stake"])
+            predictoor_user = prediction["user"]["id"]
+
+            if stake < 0.01:
+                continue
+
+            prediction_obj = Prediction(
+                pair_name,
+                timeframe,
+                predictedValue,
+                stake,
+                trueval,
+                timestamp,
+                source,
+                payout,
+                predictoor_user,
+            )
+            predictions.append(prediction_obj)
+
+    return predictions
+
+
+def get_all_contracts(owner_address: str, network: str) -> List[str]:
+    if network != "mainnet" and network != "testnet":
```

Comment:
Not a comment for this PR, and this won't work for testnet, but I think we can consolidate this functionality vs. `scripts/addresses.py` and `scripts/get_predictoor_info.py`

Reply:
Perhaps create a new github issue for that @idiom-bytes ?

Reply:
Done - https://github.com/oceanprotocol/pdr-backend/issues/307